src/Symfony/Bundle/Test/ApiTestAssertionsTrait.php

```diff
@@ -74,7 +74,7 @@ public static function assertJsonEquals(array|string $json, string $message = ''
             throw new \InvalidArgumentException('$json must be array or string (JSON array or JSON object)');
         }
 
-        static::assertEqualsCanonicalizing($json, self::getHttpResponse()->toArray(false), $message);
```

Comment:
```php
<?php

namespace App\Tests\Api;

use ApiPlatform\Symfony\Bundle\Test\ApiTestCase;

/**
 * @internal
 *
 * @coversNothing
 */
class CanolicalizingTest extends ApiTestCase
{
    public function testCanonicalizing(): void
    {
        static::assertEqualsCanonicalizing(
            [
                [
                    'id' => 1,
                ],
                [
                    'id' => 2,
                ],
            ],
            [
                [
                    'id' => 2,
                ],
                [
                    'id' => 1,
                ],
            ],
            'assertEqualsCanonicalizing :-(',
        );
        static::assertEqualsCanonicalizing(
            [
                'foo' => 5,
                'bar' => 10,
            ],
            [
                'bar' => 5,
                'foo' => 10,
            ],
            'assertEqualsCanonicalizing :-(',
        );
    }
}
```
Result : 
![image](https://github.com/api-platform/core/assets/16138648/21e3ae4e-75ab-4f41-b3bf-caf73462e108)

Reply:
mhh I see, would it be possible to sort the array so that the comparison works nonetheless? or do you think it's not needed (We had the problem in the past)

Reply:
We also need to check that comparing `['a', 'b', 'c']` with `['b', 'c', 'a']` fails because while JSON objects are unordered, JSON arrays are.

Reply:
Test will fail. I have mention an exemple in the related issued, with a result and comparison result : https://github.com/api-platform/core/issues/5751#issuecomment-1681723011